packages/datadog-plugin-google-cloud-pubsub/src/consumer.js

```diff
@@ -23,25 +24,31 @@ class GoogleCloudPubsubConsumerPlugin extends ConsumerPlugin {
       metrics: {
         'pubsub.ack': 0
       }
-    })
+    }, ctx)
+
     if (this.config.dsmEnabled && message?.attributes) {
       const payloadSize = getMessageSize(message)
       this.tracer.decodeDataStreamsContext(message.attributes)
       this.tracer
         .setCheckpoint(['direction:in', `topic:${topic}`, 'type:google-pubsub'], span, payloadSize)
     }
+
+    return ctx.currentStore
   }
 
-  finish (message) {
-    const span = this.activeSpan
+  bindFinish (ctx) {
+    const { message } = ctx
+    const span = ctx.currentStore.span
 
     if (!span) return
 
-    if (message.message._handled) {
+    if (message?._handled || message?.message?._handled) {
```

Comment:
Why can the property now be on the root or not?

Reply:
not necessary, removing.